src/clandinin_lab_to_nwb/brezovec/brezovecimagingextractor.py

```diff
@@ -0,0 +1,251 @@
+        ----------
+        folder_path : PathType
+            The path to the folder that contains the NIfTI image files (.nii) and configuration files (.xml).
+        stream_name: str, optional
+            The name of the recording channel.
+        """
+        self._niftifile = _get_nifti_reader()
+
+        folder_path = Path(folder_path)
+        nii_file_paths = list(folder_path.glob("*.nii"))
+        assert nii_file_paths, f"The NIfTI image files are missing from '{folder_path}'."
+
+        self.folder_path = Path(folder_path)
+
+        self._xml_root = _parse_xml(folder_path=folder_path)
+
+        assert self._determine_imaging_is_volumetric(self._xml_root), (
+            f"{self.extractor_name}Extractor is for volumetric imaging. "
+            "For single imaging plane data use BrezovecSinglePlaneImagingExtractor."
+        )
+
+        # TODO: All the checks on the channel_names, streams
+        self.stream_name = stream_name
+        streams = self.get_streams(folder_path=folder_path)
+        self._channel_names = list(streams["channel_streams"].keys())
+        assert (
+            stream_name in self._channel_names
+        ), f"The selected stream '{stream_name}' is not in the available channel stream '{self._channel_names}'!"
+
+        file_names = list(folder_path.glob("*.nii"))
+        # can't extract the names from xml --> they are listed as .ome.tif
+        # TODO: to be implemented in a way that is not dependent to the filename
+
+        channel_id = streams["channel_streams"][stream_name]
+        file_for_stream = [file.as_posix() for file in file_names if "channel_" + channel_id in file.as_posix()]
+        self.nibabel_image = self._niftifile.load(file_for_stream[0])
+
+        self._num_rows = self.nibabel_image.shape[1]
+        self._num_columns = self.nibabel_image.shape[0]
+        self._num_planes_per_channel_stream = self.nibabel_image.shape[2]
+
+        self._num_frames = self.nibabel_image.shape[-1]
+
+        sampling_frequency = self.get_sampling_frequency()
+        assert sampling_frequency is not None, "Could not determine the frame rate from the XML file."
+        self._sampling_frequency = sampling_frequency
+
+        self._times = None
+        self.xml_metadata = self._get_xml_metadata()
+
+    def get_image_size(self) -> Tuple[int, int, int]:
+        return (self._num_rows, self._num_columns, self._num_planes_per_channel_stream)
+
+    def get_num_frames(self) -> int:
+        return self._num_frames
+
+    def get_dtype(self) -> DtypeType:
+        return self.nibabel_image.get_data_dtype()
+
+    def get_sampling_frequency(self) -> float:
+        """
+        Determines the sampling rate from the difference in absolute timestamps of the first frame elements of each cycle or sequence element.
+        """
+        sampling_frequency = calculate_regular_series_rate(self.get_timestamps())
+        return sampling_frequency
+
+    def get_plane_acquisition_rate(self, cycle) -> float:
```

Comment:
Are you using this somewhere? I thought this is going to be a fall-back when the sampling rate cannot be determined.

Reply:
Ah sorry! No, it is not used yet. I initially wrote the function to see which was the plane_rate_acquisition, but I still don't know if it is useful info and where it should be saved.

Reply:
It is a good idea to remove unused things. I think we should move it out of the interface for now and maybe drop it by the end of the conversion project if we don't use. But I would like to wait a little bit more tilll we are focusing in the metadata to see if this can be useful to do that.